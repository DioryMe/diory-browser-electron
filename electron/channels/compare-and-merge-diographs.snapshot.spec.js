const fs = require('fs')
const { compareAndMergeDiographs } = require('./compare-and-merge-diographs')

// Load static json fixtures
const currentPath = './electron/channels'
const originalDiograph = JSON.parse(fs.readFileSync(`${currentPath}/diograph-existing.json`))
const newDiographSame = JSON.parse(fs.readFileSync(`${currentPath}/diograph-new-same.json`))
const newDiographAddedDiories = JSON.parse(
  fs.readFileSync(`${currentPath}/diograph-new-added-diories.json`)
)

describe('diograph-generator', () => {
  describe('compareAndMergeDiograph', () => {
    it('merges originalDiograph and newDiographSame', async () => {
      const diograph = compareAndMergeDiographs(originalDiograph, newDiographSame)
      expect(diograph).toMatchSnapshot()
    })

    it('merges originalDiograph and newDiographAddedDiories', async () => {
      const diograph = compareAndMergeDiographs(originalDiograph, newDiographAddedDiories)
      expect(diograph).toMatchSnapshot()
    })
  })
})
