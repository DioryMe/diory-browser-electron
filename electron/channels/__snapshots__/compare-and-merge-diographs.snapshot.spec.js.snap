// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diograph-generator compareAndMergeDiograph merges originalDiograph and newDiographAddedDiories 1`] = `
Object {
  "diograph": Object {
    "1b435867-6e03-40c7-837b-35d9de80ea90": Object {
      "created": "2021-02-19T08:55:20.502Z",
      "date": "2006-07-16T16:09:00.000Z",
      "id": "1b435867-6e03-40c7-837b-35d9de80ea90",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/Tampere/tampere.jpg",
      "modified": "2021-02-19T08:55:20.524Z",
    },
    "22d38249-751f-40d4-9691-09913656974f": Object {
      "created": "2019-10-27T14:40:43.879Z",
      "date": "2019-10-27T14:40:43.000Z",
      "id": "22d38249-751f-40d4-9691-09913656974f",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/some-folder/some-folder-image.jpg",
      "latitude": 61.49578611111111,
      "longitude": 23.76756111111111,
      "modified": "2020-10-20T18:11:31.605Z",
    },
    "3cd9ace1-ca77-4167-b5a0-0326633f56eb": Object {
      "created": "2020-04-10T18:44:56.378Z",
      "date": "2017-01-07T07:58:53.500Z",
      "id": "3cd9ace1-ca77-4167-b5a0-0326633f56eb",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/image-with-24-hour.jpg",
      "latitude": 57.74614055555555,
      "links": Object {
        "4_5807613915332347351.mp4": Object {
          "id": "5212ee46-1f2f-4bdb-a6a1-c093de9d134f",
        },
        "Frenkell": Object {
          "id": "ebc844f5-76c0-4b44-a14a-f8d0c2b401f7",
        },
        "Tampere": Object {
          "id": "43563e8d-9599-46b7-b011-a90450fe4c81",
        },
        "frenkell.jpg": Object {
          "id": "fa8d4b50-fc13-4801-b5a6-d8d6c86a69bd",
        },
        "image-with-24-hour.jpg": Object {
          "id": "4ae20908-7e29-49ce-a7fd-0ecb9a6292a3",
        },
        "other-image.jpg": Object {
          "id": "8977c2e4-f3ea-48f4-8057-76c6528ca434",
        },
        "some-folder": Object {
          "id": "6e87174a-bb9c-4b10-a8b3-d63e6dc3107a",
        },
        "some-image.jpg": Object {
          "id": "68373409-7309-4a55-8904-b84236455f07",
        },
        "some-text.txt": Object {
          "id": "dd3bc53a-f433-4810-8e7f-f7c6d7922ca8",
        },
        "tampere.jpg": Object {
          "id": "1b435867-6e03-40c7-837b-35d9de80ea90",
        },
      },
      "longitude": 16.65045388888889,
      "modified": "2021-02-19T08:57:22.760Z",
      "text": "example-folder",
    },
    "43563e8d-9599-46b7-b011-a90450fe4c81": Object {
      "created": "2021-02-19T08:55:32.160Z",
      "date": "2006-07-16T15:18:08.500Z",
      "id": "43563e8d-9599-46b7-b011-a90450fe4c81",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/Tampere/tampere.jpg",
      "links": Object {
        "Frenkell": Object {
          "id": "ebc844f5-76c0-4b44-a14a-f8d0c2b401f7",
        },
        "tampere.jpg": Object {
          "id": "1b435867-6e03-40c7-837b-35d9de80ea90",
        },
      },
      "modified": "2021-02-19T08:55:44.235Z",
      "text": "Tampere",
    },
    "4ae20908-7e29-49ce-a7fd-0ecb9a6292a3": Object {
      "created": "2021-01-23T11:17:19.400Z",
      "date": "2019-07-09T00:01:16.000Z",
      "id": "4ae20908-7e29-49ce-a7fd-0ecb9a6292a3",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/image-with-24-hour.jpg",
      "latitude": 68.27818055555555,
      "longitude": 14.302322222222223,
      "modified": "2021-01-23T11:17:19.416Z",
    },
    "5212ee46-1f2f-4bdb-a6a1-c093de9d134f": Object {
      "created": "2020-10-20T18:11:31.662Z",
      "id": "5212ee46-1f2f-4bdb-a6a1-c093de9d134f",
      "modified": "2020-10-20T18:11:31.718Z",
      "text": "4_5807613915332347351.mp4",
      "video": "/Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/4_5807613915332347351.mp4",
    },
    "68373409-7309-4a55-8904-b84236455f07": Object {
      "created": "2019-11-25T19:58:24.734Z",
      "date": "2019-11-25T19:58:24.000Z",
      "id": "68373409-7309-4a55-8904-b84236455f07",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/some-image.jpg",
      "modified": "2020-10-20T18:11:31.616Z",
    },
    "6e87174a-bb9c-4b10-a8b3-d63e6dc3107a": Object {
      "created": "2020-04-10T18:44:56.381Z",
      "date": "2019-10-27T14:40:43.000Z",
      "id": "6e87174a-bb9c-4b10-a8b3-d63e6dc3107a",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/some-folder/some-folder-image.jpg",
      "latitude": 61.49578611111111,
      "links": Object {
        "some-folder-image.jpg": Object {
          "id": "22d38249-751f-40d4-9691-09913656974f",
        },
      },
      "longitude": 23.76756111111111,
      "modified": "2020-10-20T18:11:31.597Z",
      "text": "some-folder",
    },
    "8977c2e4-f3ea-48f4-8057-76c6528ca434": Object {
      "created": "2020-10-20T18:11:31.575Z",
      "date": "2008-11-01T21:15:11.000Z",
      "id": "8977c2e4-f3ea-48f4-8057-76c6528ca434",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/other-image.jpg",
      "latitude": 43.464455,
      "longitude": 11.881478333333334,
      "modified": "2020-10-20T18:11:31.586Z",
    },
    "dd3bc53a-f433-4810-8e7f-f7c6d7922ca8": Object {
      "created": "2020-10-20T18:11:31.620Z",
      "id": "dd3bc53a-f433-4810-8e7f-f7c6d7922ca8",
      "modified": "2020-10-20T18:11:31.620Z",
      "text": "some-text",
    },
    "ebc844f5-76c0-4b44-a14a-f8d0c2b401f7": Object {
      "created": "2021-02-19T08:54:57.669Z",
      "date": "2006-07-16T14:27:17.000Z",
      "id": "ebc844f5-76c0-4b44-a14a-f8d0c2b401f7",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/Tampere/Frenkell/frenkell.jpg",
      "links": Object {
        "frenkell.jpg": Object {
          "id": "fa8d4b50-fc13-4801-b5a6-d8d6c86a69bd",
        },
      },
      "modified": "2021-02-19T08:55:05.123Z",
      "text": "Frenkell",
    },
    "fa8d4b50-fc13-4801-b5a6-d8d6c86a69bd": Object {
      "created": "2021-02-19T08:54:37.455Z",
      "date": "2006-07-16T14:27:17.000Z",
      "id": "fa8d4b50-fc13-4801-b5a6-d8d6c86a69bd",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/Tampere/Frenkell/frenkell.jpg",
      "modified": "2021-02-19T08:54:37.466Z",
    },
  },
  "rootId": "3cd9ace1-ca77-4167-b5a0-0326633f56eb",
}
`;

exports[`diograph-generator compareAndMergeDiograph merges originalDiograph and newDiographSame 1`] = `
Object {
  "diograph": Object {
    "22d38249-751f-40d4-9691-09913656974f": Object {
      "created": "2019-10-27T14:40:43.879Z",
      "date": "2019-10-27T14:40:43.000Z",
      "id": "22d38249-751f-40d4-9691-09913656974f",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/some-folder/some-folder-image.jpg",
      "latitude": 61.49578611111111,
      "longitude": 23.76756111111111,
      "modified": "2020-10-20T18:11:31.605Z",
    },
    "3cd9ace1-ca77-4167-b5a0-0326633f56eb": Object {
      "created": "2020-04-10T18:44:56.378Z",
      "date": "2017-01-07T07:58:53.500Z",
      "id": "3cd9ace1-ca77-4167-b5a0-0326633f56eb",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/image-with-24-hour.jpg",
      "latitude": 57.74614055555555,
      "links": Object {
        "image-with-24-hour.jpg": Object {
          "id": "4ae20908-7e29-49ce-a7fd-0ecb9a6292a3",
        },
        "other-image.jpg": Object {
          "id": "8977c2e4-f3ea-48f4-8057-76c6528ca434",
        },
        "some-folder": Object {
          "id": "6e87174a-bb9c-4b10-a8b3-d63e6dc3107a",
        },
        "some-image.jpg": Object {
          "id": "68373409-7309-4a55-8904-b84236455f07",
        },
        "some-text.txt": Object {
          "id": "dd3bc53a-f433-4810-8e7f-f7c6d7922ca8",
        },
      },
      "longitude": 16.65045388888889,
      "modified": "2021-02-19T08:57:22.760Z",
      "text": "example-folder",
    },
    "4ae20908-7e29-49ce-a7fd-0ecb9a6292a3": Object {
      "created": "2021-01-23T11:17:19.400Z",
      "date": "2019-07-09T00:01:16.000Z",
      "id": "4ae20908-7e29-49ce-a7fd-0ecb9a6292a3",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/image-with-24-hour.jpg",
      "latitude": 68.27818055555555,
      "longitude": 14.302322222222223,
      "modified": "2021-01-23T11:17:19.416Z",
    },
    "68373409-7309-4a55-8904-b84236455f07": Object {
      "created": "2019-11-25T19:58:24.734Z",
      "date": "2019-11-25T19:58:24.000Z",
      "id": "68373409-7309-4a55-8904-b84236455f07",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/some-image.jpg",
      "modified": "2020-10-20T18:11:31.616Z",
    },
    "6e87174a-bb9c-4b10-a8b3-d63e6dc3107a": Object {
      "created": "2020-04-10T18:44:56.381Z",
      "date": "2019-10-27T14:40:43.000Z",
      "id": "6e87174a-bb9c-4b10-a8b3-d63e6dc3107a",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/some-folder/some-folder-image.jpg",
      "latitude": 61.49578611111111,
      "links": Object {
        "some-folder-image.jpg": Object {
          "id": "22d38249-751f-40d4-9691-09913656974f",
        },
      },
      "longitude": 23.76756111111111,
      "modified": "2020-10-20T18:11:31.597Z",
      "text": "some-folder",
    },
    "8977c2e4-f3ea-48f4-8057-76c6528ca434": Object {
      "created": "2020-10-20T18:11:31.575Z",
      "date": "2008-11-01T21:15:11.000Z",
      "id": "8977c2e4-f3ea-48f4-8057-76c6528ca434",
      "image": "file:///Users/Jouni/Code/diory-browser-electron/electron/readers/example-folder/other-image.jpg",
      "latitude": 43.464455,
      "longitude": 11.881478333333334,
      "modified": "2020-10-20T18:11:31.586Z",
    },
    "dd3bc53a-f433-4810-8e7f-f7c6d7922ca8": Object {
      "created": "2020-10-20T18:11:31.620Z",
      "id": "dd3bc53a-f433-4810-8e7f-f7c6d7922ca8",
      "modified": "2020-10-20T18:11:31.620Z",
      "text": "some-text",
    },
  },
  "rootId": "3cd9ace1-ca77-4167-b5a0-0326633f56eb",
}
`;
